api/src/test/java/org/openmrs/module/fhir2/api/impl/FhirImmunizationServiceImplTest.java

```diff
@@ -144,6 +155,44 @@ public void saveImmunization_shouldCreateEncounterAndObsGroupWhenNewImmunization
 		    equalTo(new DateTimeType("2022-07-31T18:30:00.000Z").getValue()));
 	}
 	
+	@Test
+	public void saveImmunization_shouldSaveImmunizationWithNoteField() throws Exception {
+		FhirContext ctx = FhirContext.forR4();
+		IParser parser = ctx.newJsonParser();
+		String json = IOUtils.toString(
+		    Objects.requireNonNull(
+		        getClass().getResourceAsStream("/org/openmrs/module/fhir2/providers/immunization-note.json")),
+		    StandardCharsets.UTF_8);
+		Immunization newImmunization = parser.parseResource(Immunization.class, json);
+		Immunization savedImmunization = service.create(newImmunization);
+		Obs obs = obsService.getObsByUuid(savedImmunization.getId());
+		Map<String, Obs> members = helper.getObsMembersMap(obs);
+		assertThat(members.get(CIEL_161011).getValueText(), is("This is a test immunization note."));
+		assertThat(savedImmunization.getNoteFirstRep().getText(), is("This is a test immunization note."));
+	}
+	
+	@Test
+	public void saveImmunization_shouldNotFailIfNoteConceptIsMissingAndNoteProvided() throws Exception {
+		// Remove the note concept since @Before loads it
```

Comment:
Because this is a test, don't we know whether it exists or not? In other words, do we need the if check? I thought we would replace it with something like assert that this does not exist.